ewms_pilot/pilot.py

```diff
@@ -324,20 +324,39 @@ async def _ack_nack_finished_tasks(
             AsyncioTaskMessages: pending tasks and
             AsyncioTaskMessages: failed tasks (plus those in `previous_failed`)
     """
-    done, pending = await asyncio.wait(tasks.keys(), return_when=return_when)
```

Comment:
Is there a reason we don't just use `asyncio.wait` with a `timeout` every few seconds?  What does this new code do better?

Reply:
The new code acks sooner in the wait-for-all case

Reply:
Though I could do that by just using `asyncio.FIRST_COMPLETED` and a loop...

Reply:
[`f659477` (#44)](https://github.com/Observation-Management-Service/ewms-pilot/pull/44/commits/f659477e5c6f1c471d7992e1c18645cdc0c5ba2e)